File: DatabaseExercise/app/src/main/java/com/programming/android/sdu/databaseexercise/database/UserDao.java
package com.programming.android.sdu.databaseexercise.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

/**
 * Created by grzegorzbaczek on 10/03/2018.
 */

@Dao
public interface UserDao {

    @Query("SELECT * FROM user where uid = 1")
    User getUser();

    @Update
    int update(User user);

    @Insert
    void insert(User user);

    @Query("SELECT COUNT(*) from user")
    int countUsers();

}

File: Android 08 - Android and SQL databases/DatabaseExerciseMvvm/app/src/main/java/com/programming/android/sdu/databaseexercise/database/UserDao.kt
package com.programming.android.sdu.databaseexercise.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

/**
 * Created by grzegorzbaczek on 10/03/2018.
 */
@Dao
interface UserDao {

    @Query("SELECT * FROM user where uid = 1")
    fun getUser(): User

    @Update
    fun update(user: User): Int

    @Insert
    fun insert(user: User)

    @Query("SELECT COUNT(*) from user")
    fun countUsers(): Int
}